modules/ROOT/pages/custom-rulesets-validate-and-publish.adoc

```diff
@@ -20,16 +20,59 @@ include::anypoint-cli::partial$api-governance.adoc[tag=governance-ruleset-valida
+Use the following command to generate a documentation ZIP file for a ruleset YAML file. The resulting documentation ZIP file can then be used in an Exchange asset upload using the `--files` flag with the `docs.zip` option.
 
 include::anypoint-cli::partial$api-governance.adoc[tag=governance-document,leveloffset=+1]
 
 [[publish-ruleset]]
 == Publish the Custom Ruleset
 
-You can publish (upload) a ruleset and its documentation to Exchange using Anypoint CLI. You cannot use the Exchange UI to upload ruleset assets.
+You can publish (upload) a ruleset and its documentation to Exchange using Anypoint CLI. You can't use the Exchange UI to upload ruleset documentation.
+
+You can upload the following in a single upload command:
+
+[%header,cols="20a,40a,40a"]
+|===
+|Flag and Options|Result|Example
+|The ruleset YAML using the `--files` flag with the `"ruleset.yaml"` option.
+|This uploads the ruleset information included in the YAML file and ignores any other files detected in the same folder, such as the `exchange.json` file. You can use `--type=ruleset`. If you don't, the type is inferred from the `--files` flag `--ruleset.yaml` option.
+|`--files='{"ruleset.yaml":"mynewruleset.yaml", "docs.zip": "ruleset.doc.zip"}'`
+|The ruleset ZIP using the `--files` flag with the `"ruleset.zip"` option.
+|This uploads the ruleset information included in the YAML file and evaluates other files in the zip file, such as the `exchange.json` file. You can use `--type=ruleset`. If you don't, the type is inferred from the `--files` flag `"ruleset.zip"` option.
+|`--files='{"ruleset.zip":"mynewruleset.zip"}'`
+|The ruleset YAML and the ruleset documentation ZIP file using the `--files` flag with the `"ruleset.yaml"` and `"docs.zip"` options in the same `--files` flag. 
+|This uploads the ruleset YAML information plus the documentation for the ruleset. You must upload the documentation along with the ruleset. You cannot upload documentation separately later, but can add it manually. 
+|`--files='{"ruleset.yaml":"mynewruleset.yaml", "docs.zip": "ruleset.doc.zip"}'`
+|The ruleset ZIP and the ruleset documentation ZIP file using the `--files` flag with the `"ruleset.zip"` option with the `"docs.zip"` option in the same `--files` flag. 
+|This uploads the ruleset ZIP file information, evaluating any `exchange.json` file properties, plus the documentation for the ruleset. You must upload the documentation along with the ruleset. You cannot upload documentation separately later, but can add it manually. 
+|`--files='{"ruleset.zip":"mynewruleset.zip", "docs.zip": "ruleset.doc.zip"}'`
+|===
+
+=== Guidelines for Ruleset Exchange JSON Files
+
+You can use an `exchange.json` file to define the Exchange settings for rulesets. You can use any of the following properties:
+
+[%header,cols="35a,65a"]
+|===
+|Property|Description
+|main|The main file of your asset. Ensure that this matches the name of your ruleset when publishing a ruleset. 
```

Comment:
what does "the name of your ruleset" mean? I could publish anypoint-best-practices and have the ruleset file  called ruleset.yaml, and it would work fine.. should it state your ruleset file relative path?